### PR TITLE
Automatic thumbnail generation

### LINKED ISSUE #64: Set up thumbnail service
Set up service to generate a project's PNG thumbnail. For now, let's render the Page with the lowest ID (the initial page might be deleted).

### PR DESCRIPTION
**Pull request by cadecairos**
Fixes #64

### CODE REVIEW

**Comment by cadecairos** [comment]
I made a bunch of assumptions about the thumnail service endpoint, but this should do the job we want.

cc @thisandagain @toolness @simonwex @xmatthewx 

I need to add tests, and to confirm the screenshot endpoint's API

File: services/api/handlers/elements.js [hunk 1]
```diff
@@ -89,6 +89,11 @@ exports.patch = {
           return reply(err);
         }
 
+        var tail = request.tail('updating project thumbnail');
+        process.nextTick(function() {
```
Comment (thisandagain) [inline services/api/handlers/elements.js]: Why `process.nextTick`?

File: services/api/lib/thumbnails.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    };
+
+    return '/mobile-center-cropped/small/webmaker-desktop/' + new Buffer(url.format(urlObj)).toString('base64');
+  }
+
+  function updateThumbnail(project, url, tail) {
+    server.methods.projects.updateThumbnail(
+      [
+        JSON.stringify({
+          400: url
```
Comment (thisandagain) [inline services/api/lib/thumbnails.js]: We landed on 320px for the thumbnail width

File: services/api/handlers/elements.js (hunk 1 above)
Comment (cadecairos) [inline services/api/handlers/elements.js]: I want reply to execute before it starts the thumbnail update stuff

**Pull request merged**